jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,176 @@
+
+[Workflows][workflows] can help with building snaps in two ways:
+
+1. **Snap Store Channels** - As we mentioned in the previous section, when we upload to the Store we could optionally release at the same time. This allows us to designate specific jobs on CircleCI to deploy to specific Snap Channels. For example, the `master` branch could be used to deploy to the `edge` channel` while tagged releases could be used to deploy to the `stable` channel.
+1. **Parallelize Packing** - If your software is being packaged as a snap as well as something else, say a flatpak, .deb, .apk, etc, each package type could be placed in its own job and all run parallel. This allows your build to complete must fast than if the .deb package could start to build until the snap completed, and so on.
+
+Utilize CircleCI `workspaces` to move a generated snap file between jobs when neccessarily. Here's an example showing a snippet from the "from" job and a snippet of the "to" job:
+
+```
+... # from a job that already has the snap
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "*.snap"
+... # to the next job that needs the snap
+      - attach_workspace:
+          at: .
+...
+```
+
+Below is a complete example of how a snap package could be built on CircleCI. This same process is used the build the Snap pakcage for the [CircleCI Local CLI][local-cli-repo].
+
+
+## Full Example Config
+
+```
+workflows:
+  version: 2
+  main:
+    jobs:
+      - build
+      - publish:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
+
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: cibuilds/snapcraft:stable
+    steps:
+      - checkout
+      - run:
+          name: "Build Snap"
+          command: snapcraft
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "*.snap"
+
+  publish:
+    docker:
+      - image: cibuilds/snapcraft:stable
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: "Publish to Store"
+          command: |
+            mkdir .snapcraft
+            echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            snapcraft push *.snap --release stable
+```
+
+
+
+[snapcraft]: https://snapcraft.io/
+[snapcraft-store]: https://snapcraft.io/store
+[snapcraft-build]: https://build.snapcraft.io/
+[cibuilds-snapcraft]: https://github.com/cibuilds/snapcraft
+[snapcore-snapcraft]: https://github.com/snapcore/snapcraft/tree/master/docker
+[snap-guide-1]: https://docs.snapcraft.io/build-snaps/your-first-snap
+[blog]: https://circleci.com/blog/
+[docs]: https://circleci.com/docs/2.0/
+[private-envars]: https://circleci.com/docs/2.0/env-vars/#adding-environment-variables-in-the-app
+[workflows]: https://circleci.com/docs/2.0/workflows/
+[local-cli-repo]: https://github.com/circleci/local-cli
```

Comment:
Love the shorthand links — this should probably become the norm in all honesty. Much more readable.

Reply:
Thanks I agree. Your suggestion on 1 sentence per line is becoming more and more obvious to me. Once this is ready to ship, I will add the extra line breaks. If I wrote this with one sentence per line, this would have been so much easier to edit.

Reply:
I disagree, this is hard to review because I can't see the link target where the link title is. In general, we should use the exact title of the section or doc as the link text and use the words 'document' or 'section' so people know what is behind the link without having to click it and also know that they've hit the right thing when they get there. In this case, you have a link called 'private environment variables' that links to a document and section that doesn't use that terminology. When you link to a section called 'Adding Environment Variables in the App', use that same text as the link text, as in "Refer to the [Adding Environment Variables in the App] section of the Environment Variables document for instructions". Likewise, when you link to a different site, say 'Refer to the [local-cli GitHub repository]  for an example.'

Reply:
@michelle-luna that's a good point about reviewing the links — defining them at the bottom does make it harder to see the link text along with the link target.

I'll add this to our style guide since this has come up a few times.